x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_hooks.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { Immutable } from '../../../../../common/endpoint/types';
+import { GetPackagesResponse } from '../../../../../../ingest_manager/common/types/rest_spec';
+import { sendGetEndpointSecurityPackage } from '../store/policy_list/services/ingest';
+import { useKibana } from '../../../../common/lib/kibana';
+
+type UseEndpointPackageInfo = [
+  /** The Package Info. will be undefined while it is being fetched */
+  Immutable<GetPackagesResponse['response'][0]> | undefined,
+  /** Boolean indicating if fetching is underway */
+  boolean,
+  /** Any error encountered during fetch */
+  Error | undefined
+];
+
+/**
+ * Hook that fetches the endpoint package info
+ *
+ * @example
+ * const [packageInfo, isFetching, fetchError] = useEndpointPackageInfo();
+ */
+export const useEndpointPackageInfo = (): UseEndpointPackageInfo => {
```

Comment:
@oatkiller I know we have spoken on this before :)
I implemented this as a hook instead of Redux state ++ middleware. Let me know what you think. Right now, its really the only piece of code we have in Endpoint that does API data retrieval outside of Redux Middleware - thus based on feedback/input I can convert it to follow the normal pattern we have in place.

Reply:
@paul-tavares is the reasoning for using a hook here so that this is easier to reuse across the app without using middleware?  And the idea is that we'd replace this with an easier way to use the latest package?

Reply:
@kevinlog yeah, thats part of the idea. The other side of it is that Redux flow just seems to require too much code to implement something this simple. This took ~15 lines of code; converting it to Redux I assume would probably be around 3 times.

Reply:
I think this is fine. I haven't looked at this PR, but I would recommend writing some kind of 'unit' type test for this logic. From there, functional tests can confirm that its actually behaving.

Since I'm here, here's a few of my thoughts on react vs redux:

You mentioned that this took about 15 lines. I wouldn't recommend optimizing your design for code brevity. But if you do, I think you should add your test code to that count. I'm assuming that to unit test this, you'll need:

* some react helper code or a library to test hooks
* you'll need to mock `useKibana` and `sendGetEndpointSecurityPackage`

I think you'll also need to either mock this hook, or mock its dependencies in any test that uses this hook indirectly. In other words, if PageA uses this hook, then PageA's tests will need to mock this hook. Using a hook for side effects and business logic means that your components have both presentation logic and business logic.  For me, that is hard to think about. 

Re: not optimizing code brevity. Since I said that I don't optimize code brevity, I'll go into what I do optimize for:

* code should be easy to understand
  - impacts cost of maintenance, security, correctness of code, etc
* developers shouldn't have to think too hard about where to put new logic
  - e.g. For my design: side effects are in middleware, biz logic is in selectors, presentation logic is in react.
* it should be easy to test
  - For my design: all business logic should be in selectors, reducers, or middleware.
  - Selectors, reducers, and middleware are all pure functions and so don't require mocking.
  - Therefore, all biz logic can be testing w/o react, and shouldn't be overly coupled to react.
* should be easy to hire devs
  - We already hire devs who know react and redux, so I think we're on the same page.


When I write redux-centric UI code, one of my goals is to avoid ever needing to mock. One of my other goals is to avoid modeling my business logic in frameworks. Your team may not share these preferences (or you may not find your design to be

Reply:
Thanks @oatkiller . Always awesome to read your response and get your perspective. 
I used brevity in my response to Kevin above, where probably I should not have - it was not my intent for it show brevity but rather less moving part which simplified the code base (IMO). That being said, you make some good points (which I'm pretty sure you have told me in the past 😃 ) around the total cost of "the code".

Given that our App is Redux, I will refactor this to follow that pattern. Thanks again for the response.